native/mod_manager/mod_manager.c

```diff
@@ -794,53 +794,63 @@ static void read_remove_context(mem_t *mem, contextinfo_t *context)
     }
 }
 
+static apr_status_t insert_update_context_guard(server_rec *s, mem_t *mem, contextinfo_t *info, char *context,
+                                                int status)
+{
+    int len = strlen(context);
+    if (len > CONTEXTSZ) {
+        ap_log_error(APLOG_MARK, APLOG_WARNING, 0, s,
+                     "process_config: received context %s is too long (trimmed to 255 characters)", context);
+    }
```

Comment:
What about moving it after the remove return? Does it make sense to not inform user about removing too long contexts? (they would get it during insert)

Reply:
If you go that way I suggest returning an error to tomcat and add the corresponding error message to the protocol.
But that looks OK for the moment.

Reply:
@rhusar JFC is ok with the current state, what do you think?

Reply:
I need to refresh my memory on how this works. So what's the point of trimming the context? If we only have a map of trimmed contexts, we would not be able to make a correct load balancing decision. So that should qualify as an error and thus perhaps fail the operation. 

What was the behavior before these https://github.com/modcluster/mod_proxy_cluster/pull/234 changes went in? Did we just fail or trim?

Reply:
The behaviour from the native part was to trim (fixed size memory) without saying anything.

Reply:
Now it throws a warning once it is added. The question is whether it makes sense to warn again upon removing or not.

Reply:
> The behaviour from the native part was to trim (fixed size memory) without saying anything.

OK. So not issuing a warning on adding is clearly wrong - so kudos for finding this and the fix. I don't think we need to warn on removing. 

But I am wondering if this should not fail instead.

Reply:
Ok, so I'll move the warning after the return and merge it.

We can discuss the (change of) behavior today.

Reply:
> I don't think we need to warn on removing.

More I think about it, the removing might be actually even worse. Anyway, let's discuss on the call later.

Reply:
Well, it might end up with removing something else (same prefix of length >= CONTEXTSZ) . But then there would be conflict even earlier, wouldn't be?

> Anyway, let's discuss on the call later.

+1

Reply:
Exactly. The question is what happens when multiple contexts are trimmed to the same string. Are they duplicated in the registry, error or only 1 remains?

Reply:
It depends on the alias sent along the context. If it is the same, nothing is added and it is treated as the already existing one. If the context is different, new virtual host is added with the same context but different alias. There's never an error, so this PR adds at least a warning.

Reply:
https://github.com/modcluster/mod_proxy_cluster/blob/main/native/common/common.c#L443

So basically the context WILL never be routed by mod_proxy_balancer.

Reply:
I ll resolve this and continue on https://issues.redhat.com/browse/MODCLUSTER-822